qiskit_addon_sqd/fermion.py

```diff
@@ -66,6 +79,351 @@ def load(cls, filename):
+    """The SCI energy."""
+
+    sci_state: SCIState
+    """The SCI state."""
+
+    orbital_occupancies: tuple[np.ndarray, np.ndarray]
+    """The average orbital occupancies."""
+
+    rdm1: np.ndarray | None = None
+    """Spin-summed 1-particle reduced density matrix."""
+
+    rdm2: np.ndarray | None = None
+    """Spin-summed 2-particle reduced density matrix."""
+
+
+def diagonalize_fermionic_hamiltonian(
+    one_body_tensor: np.ndarray,
+    two_body_tensor: np.ndarray,
+    bit_array: BitArray,
+    samples_per_batch: int,
+    norb: int,
+    nelec: tuple[int, int],
+    *,
+    num_batches: int = 1,
+    energy_tol: float = 1e-8,
+    occupancies_tol: float = 1e-5,
+    max_iterations: int = 100,
+    sci_solver: Callable[
+        [list[tuple[np.ndarray, np.ndarray]], np.ndarray, np.ndarray, int, tuple[int, int]],
+        list[SCIResult],
+    ]
+    | None = None,
+    symmetrize_spin: bool = False,
+    include_configurations: list[int] | tuple[list[int], list[int]] | None = None,
+    initial_occupancies: tuple[np.ndarray, np.ndarray] | None = None,
+    carryover_threshold: float = 1e-4,
+    callback: Callable[[list[SCIResult]], None] | None = None,
+    seed: int | np.random.Generator | None = None,
+) -> SCIResult:
+    """Run SQD.
+
+    Args:
+        one_body_tensor: The one-body tensor of the Hamiltonian.
+        two_body_tensor: The two-body tensor of the Hamiltonian.
+        bit_array: Array of sampled bitstrings. Each bitstring should have both the
+            alpha part and beta part concatenated together, with the alpha part
+            concatenated on the right-hand side.
+        samples_per_batch: The number of bitstrings to include in each subsampled batch
+            of bitstrings.
+        norb: The number of spatial orbitals.
+        nelec: The numbers of alpha and beta electrons.
+        num_batches: The number of batches to subsample in each configuration recovery
+            iteration. This argument indirectly controls the dimensions of the
+            diagonalization subspaces. A higher value will yield larger subspace dimensions.
+        energy_tol: Numerical tolerance for convergence of the energy. If the change in
+            energy between iterations is smaller than this value, then the configuration
+            recovery loop will exit, if the occupancies have also converged
+            (see the ``occupancies_tol`` argument).
+        occupancies_tol: Numerical tolerance for convergence of the average orbital
+            occupancies. If the maximum change in absolute value of the average occupancy
+            of an orbital between iterations is smaller than this value, then the
+            configuration recovery loop will exit, if the energy has also converged
+            (see the ``energy_tol`` argument).
+        max_iterations: Limit on the number of configuration recovery iterations.
+        sci_solver: Selected configuration interaction solver function.
+
+            Inputs:
```

Comment:
Docstring looks great, did you build the docs and make sure this all renders nicely?

Reply:
I ran
```bash
tox -e docs
```
but got
```
...
  File "/home/kevinsung/projects/qiskit-addon-sqd/.tox/docs/lib/python3.10/site-packages/reno/scanner.py", line 1005, in get_notes_by_version
    raise ValueError(
ValueError: earliest-version set to unknown revision '0.4.0
```